crates/forge/bin/cmd/test/mod.rs

```diff
@@ -776,12 +843,18 @@ async fn test(
             "{}",
             format_aggregated_summary(num_test_suites, total_passed, total_failed, total_skipped)
         );
+
+        if summary {
```

Comment:
wondering, can we move all the work for building and displaying the table to this `if` as it's only necessary if it's actually going to be displayed?

Reply:
This would require storing all the test suite name and results to a separate object inside the loop and then later read inside the if block?  I wanted to avoid adding new variable/storage. 

We have to do the same to display the contract names in sorted order as well I guess. What do you guys think? Do you suggest storing the test results?  @mds1 @Evalir @mattsse